auth-api/src/main/java/io/stargate/auth/model/Credentials.java

```diff
@@ -59,7 +59,7 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof Credentials)) {
```

Comment:
The new logic admits sub-classes, while the old logic required the instance to be exactly of type `Credentials`

Reply:
Does ErrorProne require this change? What's the rationale?

Reply:
https://errorprone.info/bugpattern/EqualsGetClass

The bottom line is that there's not really a good way to make equals work with subclasses. But it's kind of a moot point here anyway since none of these classes have subclasses.